bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java

```diff
@@ -551,6 +539,23 @@ public ListenerRegistry getListenerRegistry() {
         return listenerRegistry;
     }
 
+    private static int getPoolSize() {
+        try {
+            if (poolSizeConf != null) {
+                poolSize = Integer.parseInt(poolSizeConf);
+            }
+        } catch (Throwable t) {
+            // Log and continue with default
+            getErrorStream().println("ballerina: error occurred in scheduler while reading system variable:" +
+                    RuntimeConstants.BALLERINA_MAX_POOL_SIZE_ENV_VAR + ", " + t.getMessage());
+        }
+        return poolSize;
+    }
+
+    private static PrintStream getErrorStream() {
+        return System.err;
```

Comment:
Do we need sperate method for this? Can we directly use System.err

Reply:
Moved to a static field with 5aa13c7